### ISSUE TITLE
OpenAPI analyzers for Problem Details

### ISSUE THREAD

**Issue by StevenLiekens** [dotnet/aspnetcore#39750]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

Hello, I enabled the OpenAPI analyzers for my web API and it reports undocumented status codes when I use `BadRequest()`.

``` csharp
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Mvc;

[ApiController]
[Route("things")]
public class ThingsController : ControllerBase
{
    [HttpPost("number-guess")]
    [ProducesResponseType(200)]
    public ActionResult<string> GuessNumber([Required] int number)
    {
        if (number != 4)
        {
            return BadRequest("that was not my number");
//          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Action method returns undeclared status code '400'.
        }

        return Ok("you guessed my number");
    }
}
```

 Now I want to use standardized Problem Details responses in my API. However, the analyzers don't show a warning for the same action if I use `ValidationProblem()` or `Problem()`.

### Describe the solution you'd like

Add OpenAPI analyzers for Problem Details shorthand methods.

``` csharp
return ValidationProblem(title: "that was not my number";
```

``` csharp
return Problem(title: "that was not my number", statusCode: 400);
```

**Comment by pranavkm** [comment]
@captainsafia is this a dupe of https://github.com/dotnet/aspnetcore/issues/6061?

**Comment by captainsafia** [comment]
Yep! @StevenLiekens We resolved this issue in 7.0-preview1. You can try the nightly release via https://github.com/dotnet/installer or wait until the official release drops.
